tmt/steps/common.py

```diff
@@ -0,0 +1,93 @@
+import json
+import os
+
+import tmt
+import tmt.options
+import tmt.steps
+import tmt.steps.execute
+import tmt.utils
+from tmt.base import Test
+from tmt.steps import Step, StepData
+from tmt.steps.execute import TMT_REBOOT_SCRIPT, ExecutePlugin
+from tmt.steps.provision import Guest
+
+
+class RebootCommon(ExecutePlugin):
```

Comment:
Being derived from `ExecutePlugin` does not "smell" right, it might bring in code that belongs to `tmt.steps.execute` package, like plugin discovery, for example. Does it have any actual dependency on *execute* plugin base class? Maybe a `tmt.steps.Plugin` would be good enough.

Reply:
Yes, I understand your point. I tried to limit it to `tmt.steps.Plugin`, but it is not enough. Current solution reuses and moves existing code, which comes from execute package and this is outcome.

Reply:
Then I'd say the code does not belong to the `ExecutePlugin` at all. I'm really worried this may have an undesirable effect on `RebootCommon` behavior, I'd argue it's time to make the state of things official by "liberating" the reboot code, making it a stand-alone piece of functionality `ExecutePlugin` and others would call, rather than own. A set of functions, a class to handle things, or a mixin class to bring the functionality into other classes, something like this. It doesn't have to be derived from `Plugin` class at all. But it's obvious `ExecutePlugin` is not the sole owner of reboots anymore.

Reply:
> I'm really worried this may have an undesirable effect on RebootCommon behavior, I'd argue it's time to make the state of things official by "liberating" the reboot code, making it a stand-alone piece of functionality ExecutePlugin and others would call, rather than own.

OK. I see. Actually I picked this issue as "good first issue" and didn't expect similar changes would be necessary. But what you wrote is true and I had similar impression just at the beginning. So... let's do it as you proposed :-)

> A set of functions, a class to handle things, or a mixin class to bring the functionality into other classes, something like this. It doesn't have to be derived from Plugin class at all. But it's obvious ExecutePlugin is not the sole owner of reboots anymore.

Yes, this sounds reasonable. I will find a way how to re-design it.

Reply:
Thanks for understanding, I appreciate it.